contracts/staking/DefaultTargetDispenserL2.sol

```diff
@@ -329,14 +332,31 @@ abstract contract DefaultTargetDispenserL2 is IBridgeErrors {
     ///         - Token transfer succeeds, message fails: call this function;
     ///         - Token transfer fails, message succeeds: re-send OLAS to the contract (separate vote).
     /// @param data Bytes message data that was not delivered from L1.
-    function processDataMaintenance(bytes memory data) external {
+    /// @param updateWithheldAmount True, if withheld amount update is required.
+    function processDataMaintenance(bytes memory data, bool updateWithheldAmount) external {
         // Check for the contract ownership
         if (msg.sender != owner) {
             revert OwnerOnly(msg.sender, owner);
         }
 
-        // Process the data
-        _processData(data);
+        // Process the data and calculate deposited amounts
+        uint256 totalAmount = _processData(data);
+
+        // Update withheld amount
+        if (updateWithheldAmount) {
+            uint256 localWithheldAmount = withheldAmount;
+
+            // Check for overflow
+            if (totalAmount > localWithheldAmount) {
+                revert Overflow(totalAmount, localWithheldAmount);
+            }
+
+            // Update withheld amount
+            localWithheldAmount -= totalAmount;
+            withheldAmount = localWithheldAmount;
+
+            emit WithheldAmountUpdated(localWithheldAmount);
```

Comment:
Calculate total amount being deposited from the balance of this contract. If the balance is used that is recorded in `withheldAmount` value, then it's the exact amount we need to subtract from `withheldAmount`.